src/edu/csus/ecs/pc2/ui/SubmitRunPane.java

```diff
@@ -493,14 +493,44 @@ protected void testOrSubmitRun(boolean submitTheRun) {
         if (submitTheRun) {
             try {
                 String confirmQuestion = "<HTML><FONT SIZE=+1>Do you wish to submit run for<BR><BR>" + "Problem:  <FONT COLOR=BLUE>" + Utilities.forHTML(problem.toString()) + "</FONT><BR><BR>"
-                        + "Language:  <FONT COLOR=BLUE>" + Utilities.forHTML(language.toString()) + "</FONT><BR><BR>" + "File: <FONT COLOR=BLUE>" + Utilities.forHTML(filename)
-                        + "</FONT><BR><BR></FONT>";
+                        + "Language:  <FONT COLOR=BLUE>" + Utilities.forHTML(language.toString()) + "</FONT><BR><BR>";
+
+                if (additionalFilesMCLB.getRowCount() > 0) {
+                    confirmQuestion += "<TABLE><TBODY><TR><TD><FONT SIZE=+1>Main File:</FONT></TD><TD><FONT COLOR=BLUE SIZE=+1>" + Utilities.forHTML(filename)
+                        + "</FONT></TD></TR>";
+                    String otherfilename = (String) additionalFilesMCLB.getRow(0)[0];
+                    confirmQuestion += "<TR><TD><FONT SIZE=+1>Additional File(s):</FONT></TD><TD><FONT COLOR=BLUE SIZE=+1>" + Utilities.forHTML(otherfilename) + "</FONT></TD></TR>";
+                    for (int i = 1; i < additionalFilesMCLB.getRowCount(); i++) {
+                        otherfilename = (String) additionalFilesMCLB.getRow(i)[0]; 
+                        confirmQuestion += "<TR><TD> </TD><TD><FONT COLOR=BLUE SIZE=+1>" + Utilities.forHTML(otherfilename) + "</FONT></TD></TR>";
+                    }
+                    confirmQuestion += "</TBODY></TABLE>";
+                }
+                else {
+                    confirmQuestion += "<TABLE><TBODY><TR><TD><FONT SIZE=+1>Main File:</FONT></TD><TD><FONT COLOR=BLUE SIZE=+1>" + Utilities.forHTML(filename)
+                        + "</FONT></TD></TR></TBODY></TABLE>";
```

Comment:
These two lines are essentially identical to lines 499/500 (with the exception of adding the `</TBODY><TABLE>` in line 511)  -- meaning the same code is being executed unconditionally (i.e. for the `if(true)` condition as well as for the `else` condition).   It seems to me it would be better to have these instructions appear just once -- i.e. code it as
```java
confirmQuestion += "...html which is _always_ added" ;
if (additionalFilesMCLB.getRowCount() > 0) {
  ... code to add each additional file name ...
}
confirmQuestion += "</TBODY></TABLE>";
```

However, I grant that this is somewhat in the realm of "coding style", and I certainly don't claim that the way it's done is _wrong_.  I'm also not saying it is a reason to reject the PR -- just pointing it out.

Reply:
Yes agreed, it could be made more concise.